core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -23,11 +23,16 @@
 import java.util.Map;
 import java.util.Set;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.view.ImmutableViewMetadata;
+import org.apache.iceberg.view.ViewVersion;
 
-/** Represents a change to table metadata. */
+/** Represents a change to table/view metadata. */
 public interface MetadataUpdate extends Serializable {
   void applyTo(TableMetadata.Builder metadataBuilder);
 
+  default void applyTo(ImmutableViewMetadata.Builder viewMetadataBuilder) {}
```

Comment:
Is it possible to not attach this to an Immutable interface? I'd rather have an interface that we have more control over.

Reply:
I think this should throw `UnsupportedOperationException` and complain that the update cannot be applied to a view. Otherwise, incorrect use would fail silently.

Reply:
@nastra, what is the benefit of Immutables here?

It looks like Immutables gives us a builder that doesn't do what we want (hide the complexity) and forces us to put logic about record keeping in the caller rather than in the builder. That's a major problem... so I'm skeptical that the right solution is to use more Immutables and have _more_ builders that add the functionality that we want.

Why don't we just scrap immutables here? It's been causing a lot of trouble.

Reply:
That's a fair and valid point and for builders where we'd need to have more custom logic we might actually not use generated builders and write the builder ourselves. Going forward I think there are two options:
1) leave `ViewMetadata` as Immutable but disable generating a builder. This would still perform all validation checks when a `ViewMetadata` instance is being created. This approach is currently implemented in this PR.
2) make `ViewMetadata` an actual class and implement it the same way as `TableMetadata`. This approach is implemented in https://github.com/nastra/iceberg/commit/01fd4fca5e53a371d48ba0540be449ba286eb9f2.

Let me know which one you'd prefer and I'll update this PR. For both cases we'd have our own Builder implementation.